isisd/isis_cli.c

```diff
@@ -152,14 +171,17 @@ DEFPY_YANG(ip_router_isis, ip_router_isis_cmd, "ip router isis WORD$tag",
 	ifp = nb_running_get_entry(NULL, VTY_CURR_XPATH, false);
 	if (ifp)
 		area = isis_area_lookup(tag, ifp->vrf_id);
-
 	if (!area) {
+		vrf = vrf_lookup_by_id(ifp->vrf_id);
```

Comment:
`ifp->vrf_id` will always be `0` when reading _isisd.conf_, which means this won't work as expected except when using the integrated config model (in which case the config is parsed after interface information is received from zebra). I believe this is what is causing your topotest to fail.

To fix this problem I think you need to add a "vrf" parameter to this command as well (e.g. "ip router isis 1 vrf RED").

Reply:
Thanks, added the new command "ip router isis 1 vrf RED". It is resolved, topotest is passing.